File: Szakdolgozat/Szakdolgozat/Classes/Comment.cs
using System;

namespace Szakdolgozat.Classes
{
    public class Comment : BaseClass
    {
        private int topicId;
        private int userId;
        private string text;
        private DateTime dateTime;
        private int points;
        private string user;
        private int myVote;
        private int myVoteId;
        private int[] reaction = new int[3];
        private int myReactioId;
        private int myReactionValue;


        public Comment(int id, int topicId, int userId, string text, string Uid, DateTime dateTime, string user, int points, int myVote, int myVoteId, int[] reaction, int myReactioId, int myReactionValue) : base(id, Uid)
        {
            TopicId = topicId;
            UserId = userId;
            Text = text;
            DateTime = dateTime;
            Points = points;
            User = user;
            MyVote = myVote;
            MyVoteId = myVoteId;
            Reaction = reaction;
            MyReactioId = myReactioId;
            MyReactionValue = myReactionValue;
        }

        public int TopicId { get => topicId; set => topicId = value; }
        public int UserId { get => userId; set => userId = value; }
        public string Text { get => text; set => text = value; }
        public DateTime DateTime { get => dateTime; set => dateTime = value; }
        public int Points { get => points; set => points = value; }
        public string User { get => user; set => user = value; }
        public int MyVote { get => myVote; set => myVote = value; }
        public int MyVoteId { get => myVoteId; set => myVoteId = value; }
        public int[] Reaction { get => reaction; set => reaction = value; }
        public int MyReactioId { get => myReactioId; set => myReactioId = value; }
        public int MyReactionValue { get => myReactionValue; set => myReactionValue = value; }
    }
}

File: Szakdolgozat/Szakdolgozat/ClientApp/src/containers/Comment.js
import React from 'react'
import {Button, Card, Col, Container, Row, Table} from "react-bootstrap";
import parse from 'html-react-parser';
import Vote from "./Rection/Vote";
import Reactions from "./Rection/Reactions";

class Comment extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            text: this.props.text,
            user: this.props.user,
            points: this.props.points,
            dateTime: this.props.dateTime,
            action: this.props.action,
            userId: this.props.userId,
            conId: this.props.conId,
            uid: this.props.uid,
            value: this.props.value,
            reactionNumber: this.props.reactionNumber,
            radioValue: this.props.radioValue,
            voteId: this.props.voteId,
            reactionId: this.props.reactionId
        }
    }

    render() {
        const myStyle = {
            marginBottom: "20px"
        }
        console.log(this.state.radioValue)
        return (
            <>
                <Card style={myStyle}>
                    <Card.Header className="text-center">
                        <Container fluid>
                            <Row>
                                <Col xs={1}>
                                    <Vote number={this.state.points} userId={this.state.userId}
                                          conId={this.state.conId} type={"Comment"} value={this.state.value}
                                          uid={this.state.uid}
                                          id={this.state.voteId}/>
                                </Col>
                                <Col>
                                    {this.state.user}
                                </Col>
                                <Col>
                                    {this.dateReplace(this.state.dateTime)}
                                </Col>
                                <Col xs={1}>
                                    {this.state.action}
                                </Col>
                            </Row>
                        </Container>
                    </Card.Header>
                    <Card.Body>
                        <Card.Text>
                            {parse(this.state.text)}
                            <div className="d-flex flex-row-reverse">
                                <div className="p-2">
                                    <Reactions reactionNumber={this.state.reactionNumber}
                                               radionValue={this.state.radioValue}
                                               id={this.state.reactionId}
                                               userId={this.state.userId}
                                               conId={this.state.conId} type={"Comment"} value={this.state.value}
                                               uid={this.state.uid}
                                               id={this.state.voteId}/>
                                </div>
                            </div>
                        </Card.Text>
                    </Card.Body>
                </Card>
            </>
        )
    }

    dateReplace(dateTime) {
        return dateTime.replace("T", " ");
    }
}

export default Comment